examples/functions/build_resource_id/function.tf

```diff
@@ -0,0 +1,9 @@
+locals {
+  parent_id = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup"
+  resource_type = "Microsoft.Network/virtualNetworks"
+  name = "myVNet"
+}
+
+output "resource_id" {
+  value = provider::azapi::build_resource_id(local.parent_id, local.resource_type, local.name)
```

Comment:
Please add a comment which shows the result of the provider function, it will help users understand how it works. And please also update other places.